scrabble_cheater.py

```diff
@@ -0,0 +1,767 @@
+    rack: list of single-character strings
+        The letters in the player's rack in any order.
+
+    Returns:
+    --------
+    playable_words: dictionary with string keys and list values
+        Each key string starts with 'r' or 'c' for row or column respectively
+        and which is followed by a number specifing the row or column index.
+        The list contains all of the words that can be played with the
+        letters on the board and in the rack.
+
+    """
+    potential_words = {}
+    left_most_square = min(occupied_squares, key=lambda x: x[1])
+    right_most_square = max(occupied_squares, key=lambda x: x[1])
+    top_square = min(occupied_squares, key=lambda x: x[0])
+    bottom_square = max(occupied_squares, key=lambda x: x[0])
+
+    left_boundary = left_most_square[1]
+    if left_boundary > 0:
+        left_boundary -= 1
+    right_boundary = right_most_square[1]
+    if right_boundary < BOARD_SIZE - 1:
+        right_boundary += 1
+
+    for column in range(left_boundary, right_boundary+1):
+        key = "c" + str(column)
+        potential_words[key] = []
+        rack_copy = rack[:]
+        for row in range(BOARD_SIZE):
+            letter = scrabble_board[row][column]
+            if letter:
+                rack_copy.append(letter)
+        possible_words = find_words(rack_copy, dictionary)
+        potential_words[key].extend(possible_words)
+
+    top_boundary = top_square[0]
+    if top_boundary > 0:
+        top_boundary -= 1
+    bottom_boundary = bottom_square[0]
+    if bottom_boundary < BOARD_SIZE - 1:
+        bottom_boundary += 1
+    for row in range(top_boundary, bottom_boundary+1):
+        key = "r" + str(row)
+        potential_words[key] = []
+        rack_copy = rack[:]
+        for column in range(BOARD_SIZE):
+            letter = scrabble_board[row][column]
+            if letter:
+                rack_copy.append(letter)
+        possible_words = find_words(rack_copy, dictionary)
+        potential_words[key].extend(possible_words)
+    playable_words = []
+    for vector, words in potential_words.iteritems():
+        for word in words:
+            r = []
+            result = fit_word(word, rack, vector)
+            playable_words.extend(result)
+    playable_words = sort_by_score(playable_words)
+    return playable_words[:10]
+
+def sort_by_score(playable_words):
+    words_by_score = sorted(playable_words, key=lambda word: word['score'], reverse=True)
+    return words_by_score
+
+def get_indices_of_letters(location):
+    """
+    Returns all of the letters in the vector (row or column) specified by
+    location along with their indices.
+
+    Parameters:
+    -----------
+    location: string
+
+    Returns:
+    --------
+    letters_and_indices: list of tuples
+        Each tuple has the form (character, index) where character is a letter
+        in the alphabet and index is its position in the vector.
+
+    """
+    letters_and_indices = []
+    if location[0] == 'r':
```

Comment:
cryptic stuff like this is why you want to avoid having this "convention" to denote a vector of letters